interaction-manager/src/lib_components/spatial-navigation.js

```diff
@@ -130,9 +130,54 @@ class SpatialNavigation {
 
         if (domElements.length === 0) return console.error(`${navArea.elements.join(', ')} are either not a correct selectors or the elements are not present in the DOM.`);
 
-        if (!this.navigatableElements[navArea.area]) this.navigatableElements[navArea.area] = [];
+        if (!this.navigatableElements[navArea.area]) {
+            this.navigatableElements[navArea.area] = { elements: [], distance: 0 };
+        }
+
+        this.setNavigationAreaProperties(navArea.area, domElements);
+    }
+
+    /**
+     * @param {string} area - The area to set the properties for
+     * @param {HTMLElement[]} domElements - The elements to be added to the area
+     */
+    setNavigationAreaProperties(area, domElements) {
+        this.navigatableElements[area].elements.push(...domElements);
+        this.navigatableElements[area].distance = this.getElementsDistance(this.navigatableElements[area].elements);
+        this.navigatableElements[area].overflow = this.setOverflowValues(domElements[0].parentElement);
+    }
+
+    /**
+     * Calculates the distance between the provided elements and return the max distance
+     * @param {HTMLElement[]} elements
+     * @returns {number} - The max distance between the elements
+     */
+    getElementsDistance(elements) {
+        const distances = elements.map((el) => {
+            const { x, y } = el.getBoundingClientRect();
+            return Math.hypot(x, y);
+        });
+
+        return Math.max(...distances);
+    }

```

Comment:
Can we use `reduce` here to do the same. It would be something like:

```suggestion
    getElementsDistance(elements) {
        return elements.reduce((acc, el) => {
            const { x, y } = el.getBoundingClientRect();
            const distance = Math.hypot(x, y);
            acc = acc < distance ? distance : acc;
        }, 0);
    }

```

Reply:
resolved.